numpy/random/_generator.pyx

```diff
@@ -4150,17 +4156,32 @@ cdef class Generator:
 
         i = 0
         totsize = np.PyArray_SIZE(val_arr)
-        with self.lock, nogil:
-            while i < totsize:
-                acc = 0.0
-                for j in range(k):
-                    val_data[i+j] = random_standard_gamma(&self._bitgen,
-                                                              alpha_data[j])
-                    acc = acc + val_data[i + j]
-                invacc = 1/acc
-                for j in range(k):
-                    val_data[i + j] = val_data[i + j] * invacc
-                i = i + k
+
+        # Small alpha case: Use stick-breaking approach with beta random variates
+        if alpha_arr.max() < 1.0:
+            with self.lock, nogil:
+                while i < totsize:
+                    acc = 1.
+                    for j in range(k-1):
```

Comment:
PEP 8: Spaces around `-`  here, and around `+` in `j+1`, `i+j` and `i+j+1` below.  Technically, PEP 8 does not say there *must* be spaces around an isolated binary operator like this.  It does says one should follow the style of surrounding code ("Consistency within one module or function is the most important.").  As you can see in the old code, spaces are used about `-` and `+`, so this new could should follow that style.

Reply:
Thanks for the gentle reminder. The old code actually is not consistently using whitespaces around binary operators. 

Following your suggestion, I have fixed this for the whole `dirichlet` method as a drive by fix also.